### PR TITLE
Typescriptify thread events a bit

### PR DESCRIPTION
**Pull request by Palid**
Move stringly typed events to enum, so it's harder to make a development mistake

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/models/thread.ts [hunk 1]
```diff
@@ -20,6 +20,11 @@ import { MatrixEvent } from "./event";
 import { EventTimelineSet } from './event-timeline-set';
 import { Room } from './room';
 
+export enum THREAD_EVENTS {
```
Comment (t3chguy) [inline src/models/thread.ts:23]: we normally use Camel Casing for our enum names in the singular pluralisation

File: src/models/thread.ts [hunk 2]
```diff
@@ -20,6 +20,11 @@ import { MatrixEvent } from "./event";
...
+    ready = "Thread.ready",
+    update = "Thread.update"
```
Comment (t3chguy) [inline src/models/thread.ts:25]: We normally use Camel Casing for our enum member naming

Comment (Palid) [reply src/models/thread.ts]: Done.

File: src/models/thread.ts (hunk 1 above)
Comment (Palid) [reply src/models/thread.ts]: Done.

File: src/models/thread.ts [hunk 3]
```diff
@@ -20,6 +20,11 @@ import { MatrixEvent } from "./event";
 import { EventTimelineSet } from './event-timeline-set';
 import { Room } from './room';
 
+export enum ThreadEvents {
```
Comment (t3chguy) [inline src/models/thread.ts:23]: normally we'd use `ThreadEvent.Ready` etc, rather than pluralising

File: src/models/thread.ts [hunk 4]
```diff
@@ -73,7 +78,7 @@ export class Thread extends EventEmitter {
         if (this.ready) {
             this.client.decryptEventIfNeeded(event, {});
         }
-        this.emit("Thread.update", this);
+        this.emit(ThreadEvents.Ready, this);
```
Comment (t3chguy) [inline src/models/thread.ts:81]: this seems to have gone from update to ready, is this an intentional change? definitely doesn't look like straight TSification

File: src/models/thread.ts [hunk 5]
```diff
@@ -193,4 +198,25 @@ export class Thread extends EventEmitter {
...
+        return this;
+    }
+    public addListener(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.addListener(event, listener);
+        return this;
+    }
+    public removeListener(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.removeListener(event, listener);
+        return this;
+    }
```
Comment (t3chguy) [inline src/models/thread.ts:221]: Wonder if we can do this with a generic base class or something, as this is kinda boilerplatey, e.g `EventEmitter<ThreadEvents>` etc

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/models/thread.ts (hunk 5 above)
Comment (Palid) [reply src/models/thread.ts:221]: I think we can definitely do this. Is it useful to do it within this PR?

File: src/models/thread.ts (hunk 4 above)
Comment (Palid) [reply src/models/thread.ts:81]: That's a mistake, thanks for catching! Probably went a bit too far with replacements.

File: src/models/thread.ts (hunk 5 above)
Comment (t3chguy) [reply src/models/thread.ts:221]: Up to you, I already approved

**Pull request merged**